pkgs/dart_flutter_team_lints/lib/analysis_options.yaml

```diff
@@ -35,8 +37,10 @@ linter:
     - lines_longer_than_80_chars
     - omit_local_variable_types
     - prefer_asserts_in_initializer_lists
+    - prefer_const_constructors
```

Comment:
@sigmundch - is there any risk of degrading IPL or any other negative impacts by making too many values `const` in widely used packages?

Reply:
I don't have a strong opinion with this lint, but note that we do have it enabled in 22 of 29 sampled dart-lang packages, and flutter turns this lint on.

Reply:
Indeed, it moves initialization of each of those values to be done eagerly during the IPL as you noted (as opposed to being lazy or on-demand at the time you need it.). The question comes down to how many constants we are talking about. If it's just a handful of them, it's fine, but if all the sudden this will increase the total number of constants by 2x, that's more concerning.

cc @rakudrama, although he is about to be on PTO for a while.

Reply:
It will degrade IPL but a few small constants will not make much difference. It is really hard to give advice here since it depends a lot on the number and qualities of the constants. A few moderate sized class constants would be fine, unless one of their fields were initialized with, say, a gargantuan map.

I'd like to make dart2js be smarter about the impact of larger constants on IPL (e.g. https://github.com/dart-lang/sdk/issues/32943) but I don't think it will ever be zero.

I would say go ahead since constant widgets, being identical each frame, help the flutter framework more efficiently detect parts of the widget tree that are invariant. It can translate to better frame rate.